src/app/catalogue-search/catalogue-search-listing/catalogue-search-listing.component.ts

```diff
@@ -68,6 +72,12 @@ export class CatalogueSearchListingComponent implements OnInit {
     });
   }
 
+  onPageChange() {
+    this.parameters.page = this.paginator.pageIndex;
+    this.parameters.pageSize = this.paginator.pageSize;
+    this.performSearch();
```

Comment:
This works fine, but it wasn't what I was expecting. I apologise if my intentions were not clear in the issue description.

I've worked on a lot of e-commerce websites which generally have this pattern of displaying search results/listings. What I learned from that is the URL to the page should explain exactly what the search should do, which is why you usually see very long query strings on search pages. The reasoning behind this is because of use cases like:

1. A user might bookmark that page to come back to later
2. Someone might copy the link to that search results page and send to someone else in a message/email
3. SEO reasons (though probably not applicable to MDM)
4. etc

So if there was a URL `https://my.mauro.org/#/search/listing?search=test&page=2&size=20` then this should always return the 2nd page of search results for "test", showing the next 20 results. This is how I implemented the MDE search page, but this version in `mdm-ui` does not do that because each time you change pages the URL is not updated. At the moment, if I passed in a query param of `page=2`, I would still initially see the first page of results.

This is easily fixed though. If instead of re-rendering the search results list, you instead could use the UI router to navigate to the same page route but with the updated query params - see the `SearchListingComponent` in `mdm-explorer` for what I mean. This should then render a new `CatalogueSearchListingComponent` and trigger the `ngOnInit()` to start the next search call.

Because `mdm-ui` uses UI Router though instead of Angular router, I can't remember if just transitioning the page route is enough, you might need to also pass a `{ reload: true }` option to the router to force a reload - something you'll have to test.

Reply:
Thanks for the info. As well as changing the route to accept an additional `pageSize` query parameter, I also had to change `MdmPaginatorComponent` to:
(a) accept a `[pageIndex]` input
(b) properly deal with the `[pageSize]` input i.e. rather than always setting pageSize to 20, use the pageSize from the input, or failing that from user settings, else default to 20.